src/any.rs

```diff
@@ -578,6 +593,15 @@ impl Dynamic {
             Ok(s) => return (*s).into(),
             Err(val) => val,
         };
+
+        #[cfg(feature = "decimal")]
+        {
+            boxed = match unsafe_cast_box::<_, rust_decimal::Decimal>(boxed) {
```

Comment:
I suppose `Decimal` is `Copy` so it can use `unsafe_try_cast` to save on one boxing operation?

Reply:
Maybe just copy the `FLOAT` block:

```rust
        #[cfg(not(feature = "no_float"))]
        if type_id == TypeId::of::<Decimal>() {
            return <dyn Any>::downcast_ref::<Decimal>(&value)
                .unwrap()
                .clone()
                .into();
        }
```

Reply:
Also, if you have `serde` enabled, it may attempt to serialize `f32` or `f64` into a `Dynamic` via `Dynamic::from`.  Therefore,  you'd want to also handle the case where 1) `no_float`, 2) `decimal`, 3) `serde` (which I suppose must have floats...), 4) the type is `f32` or `f64`.  In such a case, you should convert the floating-point number into `Decimal`.:

Reply:
Using `unsafe_try_cast` is possible, but this breaks the transfer of ownership pattern at play, where the `Result::Err<Box<...>` returned from `unsafe_cast_box` is used to move the boxed value back from the `unsafe_try_cast`.

Reply:
The problem being that `unsafe_try_cast` returns an `Option` meaning it consumes the boxed value. I'm unsure how to resolve that. Maybe by borrowing the contents of the Box to `unsafe_try_cast`?

Reply:
Sorry, my mistake. You can't use `unsafe_try_cast` here.  Just `downcast_ref` and then `clone` as my example.

Reply:
They key points are: 1) `Decimal` is `Copy` so cloning is cheaper than boxing (supposedly), 2) You should handle the rate conversion cases to/from f32/f64.

Reply:
`downcast_ref` version is pushed.

I have to look at the `serde` case closer later on. I need to understand the flow there. `rust_decimal` has `serde` support.

Reply:
Yes.  I think it probably defaults to serialize to/from JSON as f64 or f32.  Thus you need to consider some of the plumbing in the `serde` feature that serializes to/from `Dynamic`.

However, it is not a big deal, as I can clean them up later on.